src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -47,20 +48,22 @@ public partial class CompareInfo : IDeserializationCallback
         // The interesting part is that since haw-US doesn't have its own sort, it has to point at another
         // locale, which is what SCOMPAREINFO does.
         [OptionalField(VersionAdded = 2)]
-        private string m_name;  // The name used to construct this CompareInfo. Do not rename (binary serialization)
+        private string? m_name;  // The name used to construct this CompareInfo. Do not rename (binary serialization)
 
         [NonSerialized]
-        private string _sortName; // The name that defines our behavior
+        private string? _sortName; // The name that defines our behavior
 
         [OptionalField(VersionAdded = 3)]
-        private SortVersion m_SortVersion; // Do not rename (binary serialization)
+        private SortVersion? m_SortVersion; // Do not rename (binary serialization)
 
         private int culture; // Do not rename (binary serialization). The fields sole purpose is to support Desktop serialization.
 
         internal CompareInfo(CultureInfo culture)
         {
             m_name = culture._name;
             InitSort(culture);
+            // TODO-NULLABLE: Workaround compiler which sees this as uninitialized
```

Comment:
It still happens on the latest compiler:
https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQDk4BstWFiAEQHsBzMY4gEwBoqQBqAHwGIA7bXfCAAgjbwEAsACgAAgAYeYgIwBuUWIDM0gEw8AwjwDeonvp4AHBAEsAbrF6ypAfQAeCkQekqxAFh4BZABQBKHXrOBgCSbCYwABp+js4AvoH6CUamFjBWHqHhUb5Juk5B+vY8ALzSMhIAdACiALaGMACeMQbxIrFAA==

Maybe we should go ahead at this point and open an issue into the roslyn team.